edx_solutions_api_integration/courses/views.py

```diff
@@ -309,47 +201,71 @@ def _manage_role(course_descriptor, user, role, action):
                 update_forum_role(course_descriptor.id, user, FORUM_ROLE_MODERATOR, 'revoke')
 
 
-def _get_course_data(request, course_key, course_descriptor, depth=0):
+def _make_block_tree(request, blocks_data, course_key, course_block, block=None, depth=1, usage_key=None):
     """
-    creates a dict of course attributes
+    Its a nested method that will return a serialized details
+    of a content block and its children depending on the depth
+
+    usage_key must be provided in case of no root/parent block
     """
+    data = {}
+    children = []
 
-    if depth > 0:
-        data = _serialize_content_with_children(
-            request,
-            course_key,
-            course_descriptor,  # Primer for recursive function
-            depth
-        )
-        data['content'] = data['children']
-        data.pop('children')
+    protocol = 'http'
+    if request.is_secure():
+        protocol = protocol + 's'
+
+    base_content_uri = '{}://{}/api/server/courses'.format(
+        protocol,
```

Comment:
Can we clear the clutter around protocol and use some like this `request.scheme`?